docs/sessions.md

```diff
@@ -0,0 +1,127 @@
+  startWebServer suaveCfg app
+  0 
+{% endhighlight %}
+
+Cookie Serialization<br>_(of particular interest to .NET Core < netstandard2.0)_
+--------------------------------------------------------------------------------
+
+Suave uses the .NET Framework type `BinaryFormatter` to serialize the `Map<string, obj>` containing the session state; this is the default. However, the `BinaryFormatter` was removed in the .NET Core API, and the `DataContractJsonSerializer` does not recognize the `Map<string, obj>` type. One option is to utilize JSON.NET to serialize this object. To use that, ensure you've added the `Newtonsoft.Json` NuGet package to your project, then put the following code somewhere before the `suaveCfg` definition in the example above.
+
+{% highlight fsharp %}
+/// alias
+let utf8 = System.Text.Encoding.UTF8
+
+type JsonNetCookieSerialiser () =
+  interface CookieSerialiser with
+    member x.serialise m =
+      utf8.GetBytes (JsonConvert.SerializeObject m)
+    member x.deserialise m =
+      JsonConvert.DeserializeObject<Map<string, obj>> (utf8.GetString m)
+{% endhighlight %}
+
+Then, modify the configuration to use that serializer.
+
+{% highlight fsharp %}
+/// Again, do not make this your server key in production...
+let suaveCfg =
+  { defaultConfig with
+      serverKey = System.Text.Encoding.UTF8.GetBytes("12345678901234567890123456789012")
+      cookieSerialiser = new JsonNetCookieSerialiser()
+    }
+{% endhighlight %}
+
+State among WebParts
+--------------------
+
+Within the `Writers` module, Suave provides the functions `setUserData` and `unsetUserData` for adding items to the context's `userState` property.  The example below could be used to accrue a list of messags to be displayed to the user.
+
+{% highlight fsharp %}
+/// string -> WebPart
+let addUserMessage message =
+  context (fun ctx ->
+    let existing =
+      match ctx.userState.ContainsKey "messages" with
+      | true -> ctx.userState.["messages"] :?> string list
+      | _ -> []
+    Writers.setUserData "messages" (message :: existing))
+
+/// WebPart
+let app =
+  choose [
+    path "/"
+      >=> addUserMessage "It's a state!"
+      >=> addUserMessage "Another one"
+      >=> context (fun ctx -> Successful.OK (View.page ctx.userState))
+      >=> Writers.unsetUserData "messages"
+    ]
+{% endhighlight %}
+
+In this example, `View.page` is a function that generates the output, using the user state `Map<string, obj>` to display the messages in a nice way.  Additionally, this example shows a few other interesting aspects.  First, the user state may persist across requests; if that is not desirable for your use case, you will want to `unset` the items when they are no longer needed.  Second, you can combine `WebPart`s even after the one that sets the output content.  If you are used to an MVC environment, you can't do much after you `return` your result; these combinators let you modify the context even once the output has been generated.
```

Comment:
I know that was more for Ademar; I'll watch this conversation. A "gotcha" is one thing, but an "unintentional gotcha" is pretty much the definition of a bug.  :)  If it is determined that it is a bug, I'll axe that that `unsetUserData` line and fix the verbiage below it to explain the per request/per session lifetimes.

One consideration, though, is that with its current behavior, if people are using the post/redirect/get sequence of requests, `userData` is a spot to stash things that only need to persist in the context of this sequence. If that is changed, we'd want to make sure people understood that this unintended functionality isn't going to work the way it used to.

I stand ready to document your decision.  :)  _(...and the reasoning behind it)_